sqlserver/datadog_checks/sqlserver/statements.py

```diff
@@ -51,34 +51,56 @@
 with qstats as (
     select TOP {limit} query_hash, query_plan_hash, last_execution_time, plan_handle,
            (select value from sys.dm_exec_plan_attributes(plan_handle) where attribute = 'dbid') as dbid,
+            statement_start_offset,
+            statement_end_offset,
            {query_metrics_columns}
     from sys.dm_exec_query_stats
     where last_execution_time > dateadd(second, -?, getdate())
 ),
 qstats_aggr as (
     select query_hash, query_plan_hash, CAST(S.dbid as int) as dbid,
        D.name as database_name, max(plan_handle) as plan_handle,
+       statement_start_offset,
+       statement_end_offset,
     {query_metrics_column_sums}
     from qstats S
     left join sys.databases D on S.dbid = D.database_id
-    group by query_hash, query_plan_hash, S.dbid, D.name
+    group by query_hash, query_plan_hash, S.dbid, D.name, statement_start_offset, statement_end_offset
 )
-select text, encrypted as is_encrypted, * from qstats_aggr
-    cross apply sys.dm_exec_sql_text(plan_handle)
+select
+    SUBSTRING(text, (statement_start_offset / 2) + 1,
+    ((CASE statement_end_offset
+        WHEN -1 THEN DATALENGTH(text)
+        ELSE statement_end_offset END
+            - statement_start_offset) / 2) + 1) AS statement_text,
+    encrypted as is_encrypted,
+    (SELECT IIF (EXISTS (SELECT 1 FROM sys.dm_exec_procedure_stats WHERE object_id =qt.objectid), 1, 0)) as is_proc,
```

Comment:
the only way I could find to get a global view of what was a procedure based on `objectid` was to look it up in the `sys.dm_exec_procedure_stats`. I'm hoping this doesn't add a ton of overhead, and am open to other ways to look up this information. 

querying the `sys.objects` and `sys.all_objects` table didn't work in my testing. For example, if I ran this query from the `master` db I was not able to determine the object type for procs executed in `datadog_test`. 

functions like `OBJECTPROPERTY`, which can test if an object is a stored procedure based on objectid also only apply to schema-scoped objects in the current database.